src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs

```diff
@@ -163,51 +163,90 @@ public HttpClient(HttpMessageHandler handler, bool disposeHandler)
+
+                    // Get the response content.
+                    HttpContent? c = responseMessage.Content;
+                    if (c != null)
                     {
-                        try
+                        if (HttpTelemetry.Log.IsEnabled())
                         {
-                            await responseStream.CopyToAsync(buffer, cancellationToken).ConfigureAwait(false);
+                            HttpTelemetry.Log.ResponseContentStart();
+                            responseContentTelemetryStarted = true;
                         }
-                        catch (Exception e) when (HttpContent.StreamCopyExceptionNeedsWrapping(e))
+    #if NET46
+                        return await c.ReadAsStringAsync().ConfigureAwait(false);
+    #else
+                        HttpContentHeaders headers = c.Headers;
+
+                        // Since the underlying byte[] will never be exposed, we use an ArrayPool-backed
+                        // stream to which we copy all of the data from the response.
+                        using (Stream responseStream = c.TryReadAsStream() ?? await c.ReadAsStreamAsync(cancellationToken).ConfigureAwait(false))
+                        using (var buffer = new HttpContent.LimitArrayPoolWriteStream(_maxResponseContentBufferSize, (int)headers.ContentLength.GetValueOrDefault()))
                         {
-                            throw HttpContent.WrapStreamCopyException(e);
-                        }
+                            try
+                            {
+                                await responseStream.CopyToAsync(buffer, cancellationToken).ConfigureAwait(false);
+                            }
+                            catch (Exception e) when (HttpContent.StreamCopyExceptionNeedsWrapping(e))
+                            {
+                                throw HttpContent.WrapStreamCopyException(e);
+                            }
 
-                        if (buffer.Length > 0)
-                        {
-                            // Decode and return the data from the buffer.
-                            return HttpContent.ReadBufferAsString(buffer.GetBuffer(), headers);
+                            if (buffer.Length > 0)
+                            {
+                                // Decode and return the data from the buffer.
+                                return HttpContent.ReadBufferAsString(buffer.GetBuffer(), headers);
+                            }
                         }
+    #endif
                     }
-#endif
+
+                    // No content to return.
+                    return string.Empty;
                 }
+            }
+            catch when (LogRequestFailed(telemetryStarted))
+            {
+                // Unreachable as LogRequestFailed will return false
+                throw;
+            }
+            finally
+            {
+                if (HttpTelemetry.Log.IsEnabled() && telemetryStarted)
```

Comment:
If we're doing `ResponseContentStop` only under `telemetryStarted`, shouldn't we do the same with `ResponseContentStart`? Currently it's only guarded with [`IsEnabled`](https://github.com/dotnet/runtime/pull/41022/files#diff-256c040533e94c4e518e0453c5a7d4a0R197)

Same for other occurrences.

Reply:
Added the `telemetryStarted` check.